### PR TITLE
Instana Exporter - Initial support

### PR DESCRIPTION
**Pull request by zivaninstana**
Open Sourcing the OpenTelemetry Instana Exporter.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+	<GeneratePackageOnBuild>false</GeneratePackageOnBuild>
+	<PackageId>OpenTelemetry.Exporter.Instana</PackageId>
+	<PackageTags>Instana Tracing APM</PackageTags>
+	<Copyright>Copyright 2019-</Copyright>
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj:8]: Please cleanup this - you can copy most from any existing projects.

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry.Exporter.Instana/Implementation/InstanaSpanFactory.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("OpenTelemetry.Exporter.Instana.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010051c1562a090fb0c9f391012a32198b5e5d9a60e9b80fa2d7b434c9e5ccb7259bd606e66f9660676afc6692b8cdc6793d190904551d2103b7b22fa636dcbb8208839785ba402ea08fc00c8f1500ccef28bbf599aa64ffb1e1d5dc1bf3420a3777badfe697856e9d52070a50c3ea5821c80bef17ca3acffa28f89dd413f096f898")]
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Instana/Implementation/InstanaSpanFactory.cs:19]: do you need this per class? Some of the other projects in the repo use this approach, per project basis

File: src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net461|AnyCPU'">
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net461|AnyCPU'">
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+
+    <ItemGroup>
+        <None Include=".\icon.png" Link="icon.png">
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj:27]: do you need this?

File: src/OpenTelemetry.Exporter.Instana/TracerProviderBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Runtime.CompilerServices;
+using OpenTelemetry.Trace;
+
+[assembly: InternalsVisibleTo("OpenTelemetry.Exporter.Instana.Tests, PublicKey=002400000480000094000000060200000024000052534131000400000100010051c1562a090fb0c9f391012a32198b5e5d9a60e9b80fa2d7b434c9e5ccb7259bd606e66f9660676afc6692b8cdc6793d190904551d2103b7b22fa636dcbb8208839785ba402ea08fc00c8f1500ccef28bbf599aa64ffb1e1d5dc1bf3420a3777badfe697856e9d52070a50c3ea5821c80bef17ca3acffa28f89dd413f096f898")]
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Instana/TracerProviderBuilderExtensions.cs:21]: this class is already public, so internalsvisible is not required here.

File: opentelemetry-dotnet-contrib.sln [hunk 5]
```diff
@@ -185,6 +185,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentati
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Extensions.PersistentStorage.Abstractions", "src\OpenTelemetry.Extensions.PersistentStorage.Abstractions\OpenTelemetry.Extensions.PersistentStorage.Abstractions.csproj", "{17E3936A-265A-4C9F-9DD5-4568F80E6D91}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.Instana", "src\OpenTelemetry.Exporter.Instana\OpenTelemetry.Exporter.Instana.csproj", "{BD3C6377-6F8D-47D6-9710-1681ED4E6772}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Exporter.Instana.Tests", "test\OpenTelemetry.Exporter.Instana.Tests\OpenTelemetry.Exporter.Instana.Tests.csproj", "{77E7DDB9-32CF-450E-B596-E893149D07DD}"
+EndProject
```
Comment (utpilla) [inline opentelemetry-dotnet-contrib.sln:191]: Include the .sln changes to add the `package-Exporter.Instana.yml` as well.

Comment (zivaninstana) [reply opentelemetry-dotnet-contrib.sln:191]: Sorry, I do not understand what I have to do. Could you please give me more details?

Comment (utpilla) [reply opentelemetry-dotnet-contrib.sln:191]: Add the .yml file to this section: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/opentelemetry-dotnet-contrib.sln#L26-L52

File: src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj (hunk 1 above)
Comment (zivaninstana) [reply src/OpenTelemetry.Exporter.Instana/OpenTelemetry.Exporter.Instana.csproj]: Fixed

File: src/OpenTelemetry.Exporter.Instana/Implementation/InstanaSpanFactory.cs (hunk 2 above)
Comment (zivaninstana) [reply src/OpenTelemetry.Exporter.Instana/Implementation/InstanaSpanFactory.cs]: Fixed

File: src/OpenTelemetry.Exporter.Instana/TracerProviderBuilderExtensions.cs (hunk 4 above)
Comment (zivaninstana) [reply src/OpenTelemetry.Exporter.Instana/TracerProviderBuilderExtensions.cs]: Fixed

File: opentelemetry-dotnet-contrib.sln (hunk 5 above)
Comment (zivaninstana) [reply opentelemetry-dotnet-contrib.sln:192]: Thanks. Fixed

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@zivaninstana LGTM. Please fix the CI failures and we are good to merge.

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**